### ISSUE TITLE
v2.0.5.1 Exception on nested projects

### ISSUE THREAD

**Issue by Rycochet** [Everlaw/nbts#71]
This is a change between 2.0.5.0 and 2.0.5.1 (so I've had to go back to the earlier version to get any work done).

Not sure if it's relevant (but probably is) - nested Typescript projects, with the outermost one referencing the same files as the inner two (as a dummy that never actually gets compiled so it knows where to look for things). Using Grunt for building so NBTS is only handling the editor side of things.

After updating all of the .ts files are getting an error on the first line:

`Unknown source root for file path/to/xyz.ts`

All of the tsconfig.json files are getting this error:

`Error communicating with Node.js process`

There is also an Unexpected Exception, which looking at the two commits between releases makes me think something got missed (and is responsible for the editor warnings)?

```
java.lang.NullPointerException: The fileObject parameter cannot be null
	at org.openide.util.Parameters.notNull(Parameters.java:89)
	at org.netbeans.modules.parsing.api.Source.create(Source.java:138)
	at netbeanstypescript.TSService$NodeJSProcess.<init>(TSService.java:170)
	at netbeanstypescript.TSService.addFile(TSService.java:282)
Caused: java.lang.RuntimeException
	at netbeanstypescript.TSService.addFile(TSService.java:297)
	at netbeanstypescript.TSIndexerFactory$1.index(TSIndexerFactory.java:86)
	at org.netbeans.modules.parsing.spi.indexing.Indexable$MyAccessor$2.run(Indexable.java:161)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.runIndexer(RepositoryUpdater.java:298)
	at org.netbeans.modules.parsing.spi.indexing.Indexable$MyAccessor.index(Indexable.java:159)
[catch] at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.doIndex(RepositoryUpdater.java:2773)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.access$800(RepositoryUpdater.java:2177)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work$1.run(RepositoryUpdater.java:2659)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work$1.run(RepositoryUpdater.java:2657)
	at org.netbeans.modules.parsing.impl.indexing.errors.TaskCache.refreshTransaction(TaskCache.java:565)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.index(RepositoryUpdater.java:2657)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$AbstractRootsWork$4.call(RepositoryUpdater.java:5729)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$AbstractRootsWork$4.call(RepositoryUpdater.java:5637)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$4.run(RepositoryUpdater.java:2150)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:304)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.runInContext(RepositoryUpdater.java:2146)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.runInContext(RepositoryUpdater.java:2127)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater.access$1500(RepositoryUpdater.java:159)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$AbstractRootsWork.scanSource(RepositoryUpdater.java:5764)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$AbstractRootsWork.scanSources(RepositoryUpdater.java:5434)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$RootsWork.getDone(RepositoryUpdater.java:5053)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$InitialRootsWork.getDone(RepositoryUpdater.java:5836)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Work.doTheWork(RepositoryUpdater.java:3435)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task._run(RepositoryUpdater.java:6198)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task.access$4300(RepositoryUpdater.java:5849)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task$2$1.run(RepositoryUpdater.java:6114)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:304)
	at org.netbeans.modules.parsing.impl.RunWhenScanFinishedSupport.performScan(RunWhenScanFinishedSupport.java:106)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task$2.call(RepositoryUpdater.java:6110)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task$2.call(RepositoryUpdater.java:6106)
	at org.netbeans.modules.masterfs.filebasedfs.utils.FileChangedManager.priorityIO(FileChangedManager.java:176)
	at org.netbeans.modules.masterfs.providers.ProvidedExtensions.priorityIO(ProvidedExtensions.java:360)
	at org.netbeans.modules.parsing.nb.DataObjectEnvFactory.runPriorityIO(DataObjectEnvFactory.java:141)
	at org.netbeans.modules.parsing.impl.Utilities.runPriorityIO(Utilities.java:88)
	at org.netbeans.modules.parsing.impl.indexing.RepositoryUpdater$Task.run(RepositoryUpdater.java:6106)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1443)
	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:68)
	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:303)
	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2058)
```

**Comment by jeffrey-easyesi** [comment]
Looks like it's somehow failing to read the .d.ts files from the jar. What OS and NetBeans version are you using? Also, can you try this and see if it work: [netbeanstypescript-2.0.5.2-issue71.zip](https://github.com/Everlaw/nbts/files/557255/netbeanstypescript-2.0.5.2-issue71.zip)?

**Comment by Rycochet** [comment]
It's using TS 2.0.5, NetBeans 8.2 on Windows 7 x64.

Will try the updated build at work tomorrow :-)

**Comment by Rycochet** [comment]
@jeffrey-easyesi Can confirm that download works :-)

**Comment by jeffrey-easyesi** [comment]
OK, I've released that as 2.0.5.2.
I think this may be the original cause: https://bugs.openjdk.java.net/browse/JDK-8131067. It would only affect someone whose user directory has spaces or other non-alphanumeric characters in it, which would explain why I didn't see it in my own testing.
